### PR TITLE
Homepage slider

### PR DESCRIPTION
**Pull request by nadiyafomenko**

### CODE REVIEW

File: src/redux/root.saga.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ import checkoutSaga from './checkout/checkout.sagas';
 import chatSaga from './chat/chat.sagas';
 import headerLinksSaga from './header-links/header-links.sagas';
 import homeLooksImagesSaga from './home-page-looks/home-page-looks.sagas';
+import HomePageSliderSaga from './homepage-slider/homepage-slider.sagas';
```
Comment (DmytroDidukh) [inline src/redux/root.saga.js:17]: Please, lower case that import to according of rest above.

**Review by DmytroDidukh — COMMENTED** [review]
And please, check code smells.

File: src/redux/homepage-slider/homepage-slider.sagas.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import {
+  setHomePageSliderImages,
+  setHomePageSliderImagesLoading
+} from './homepage-slider.actions';
+import { setError } from '../error/error.actions';
+import getItems from '../../utils/client';
+import { GET_HOMEPAGE_SLIDER_IMAGES } from './homepage-slider.types';
+
+export function* handleHomePageSliderImagesLoad() {
+  console.log('images');
```
[the remark this answers is not in the document]
Comment (nadiyafomenko) [reply src/redux/homepage-slider/homepage-slider.sagas.js]: fixed

File: src/pages/home/slider-home-page/slider-home-page.js [hunk 3]
```diff
@@ -1,28 +1,37 @@
...
-    models: Model.models,
-    loading: Model.loading
+  const dispatch = useDispatch();
+
+  const { images, loading } = useSelector(({ HomePageSlider }) => ({
+    images: HomePageSlider.images,
+    loading: HomePageSlider.loading
+  }));
+
+  const { language } = useSelector(({ Language }) => ({
```
[the remark this answers is not in the document]
Comment (nadiyafomenko) [reply src/pages/home/slider-home-page/slider-home-page.js]: fixed

File: src/redux/root.saga.js (hunk 1 above)
Comment (nadiyafomenko) [reply src/redux/root.saga.js]: fixed

File: src/pages/home/slider-home-page/slider-home-page.js [hunk 4]
```diff
@@ -32,15 +39,35 @@ const SliderHomePage = () => {
...
-  return models.length > 0 ? (
+  return (
     <div className={styles.captionBlock}>
-      <AwesomeSlider className={styles.slider} mobileTouch>
-        {models.map((photo, index) => (
-          <div key={photo.name[1].value} data-src={images[index]} />
+      <AutoplaySlider
+        play
+        cancelOnInteraction
+        interval={6000}
```
Comment (TorskyM) [inline src/pages/home/slider-home-page/slider-home-page.js:47]: Add `sliderInterval` to the config file

File: src/redux/homepage-slider/homepage-slider.sagas.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+                 }          
+                }`;
+  try {
+    const images = yield call(getItems, query);
+    yield put(setHomePageSliderImages(images.data.getAllSlides));
+    yield put(setHomePageSliderImagesLoading(false));
+  } catch (e) {
+    yield put(setHomePageSliderImagesLoading(false));
+    yield put(setError({ e }));
+    yield put(push('/error-page'));
```
Comment (TorskyM) [inline src/redux/homepage-slider/homepage-slider.sagas.js:39]: Move error handling to the separate function

File: src/pages/home/slider-home-page/slider-home-page.js (hunk 4 above)
Comment (nadiyafomenko) [reply src/pages/home/slider-home-page/slider-home-page.js]: fixed

File: src/redux/homepage-slider/homepage-slider.sagas.js (hunk 5 above)
Comment (nadiyafomenko) [reply src/redux/homepage-slider/homepage-slider.sagas.js]: fixed

File: src/pages/home/slider-home-page/slider-home-page.js [hunk 6]
```diff
@@ -1,28 +1,37 @@
...
+    loading: HomePageSlider.loading
+  }));
+
+  const { language } = useSelector(({ Language }) => ({
+    language: Language.language
   }));
 
-  useMemo(() => {
-    models.forEach((item) => {
-      getImage(item.images.large)
```
[the remark this answers is not in the document]
Comment (nadiyafomenko) [reply src/pages/home/slider-home-page/slider-home-page.js:21]: fixed

**Review by TorskyM — APPROVED** [review]

**Review by ahd3r — APPROVED** [review]

**Review by DmytroDidukh — APPROVED** [review]

**Pull request merged**